elements/pl-matching/pl-matching.py

```diff
@@ -0,0 +1,399 @@
+import prairielearn as pl
+import lxml.html
+import random
+import math
+import chevron
+
+WEIGHT_DEFAULT = 1
+FIXED_ANSWER_ORDER_DEFAULT = False
+INLINE_DEFAULT = False
+PARTIAL_CREDIT_DEFAULT = True
+HIDE_ANSWER_PANEL_DEFAULT = False
+HIDE_HELP_TEXT_DEFAULT = False
+DETAILED_HELP_TEXT_DEFAULT = False
+HIDE_LETTER_KEYS_DEFAULT = False
+HIDE_SCORE_BADGE_DEFAULT = False
+BLANK_DEFAULT = False
+BLANK_ANSWER = ' '
+NOTA_DEFAULT = False
+COUNTER_TYPE_DEFAULT = 'lower-alpha'
+
+
+def get_form_name(answers_name, index):
+    return f'{answers_name}-dropdown-{index}'
+
+
+def get_counter(i, counter_type):
+    """Converts an integer counter to the specified CSS counter type"""
+    if counter_type == 'lower-alpha':
+        return chr(ord('a') + i - 1)
+    elif counter_type == 'upper-alpha':
+        return chr(ord('A') + i - 1)
+    elif counter_type == 'decimal':
+        return str(i)
+    else:
+        raise Exception('Illegal counter_type in pl-matching element.')
+
+
+def legal_answer(answer, counter_type, options):
+    """Checks that the given answer is within the range of the given counter type."""
+    if counter_type == 'lower-alpha':
+        return 'a' <= answer <= chr(ord('a') + len(options))
+    elif counter_type == 'upper-alpha':
+        return 'A' <= answer <= chr(ord('A') + len(options))
+    elif counter_type == 'decimal':
+        return 1 <= answer <= len(options)
+    else:
+        raise Exception('Illegal counter_type in pl-matching element.')
```

Comment:
Based on how this function is used, I believe it makes more sense to return false in this case.
```suggestion
        return False
```